File: Dash MPD Player/mitube.app/core/src/main/java/com/mitube/core/models/Channel.kt
package com.mitube.core.models

import com.google.gson.annotations.SerializedName

data class Channel(
    val name: String = "",
    val url: String = "",
    val type: String = "",
    val drm_license_uri: String = "",
    val icono: String = "",
    val headers: Map<String, String>? = null,
    val manifestUrl: String = "",
    val licenseProxyUrl: String = "",
    val keyId: String = "",
    val key: String = ""
) {
    data class ClearkeyLicense(val keyId: String, val key: String)

    fun parseClearkeyLicense(): ClearkeyLicense? {
        if (drm_license_uri.isBlank()) return null
        val uri = java.net.URI(drm_license_uri)
        val params = uri.query?.split("&")
            ?.mapNotNull { param ->
                val parts = param.split("=", limit = 2)
                if (parts.size == 2) parts[0] to parts[1]
                else null
            }?.toMap() ?: return null

        val keyId = params["keyid"] ?: return null
        val key = params["key"] ?: return null
        if (keyId.isBlank() || key.isBlank()) return null
        return ClearkeyLicense(keyId, key)
    }

    val isClearkey: Boolean get() = type.equals("CLEARKEY", ignoreCase = true)
    val isWidevine: Boolean get() = type.equals("WIDEVINE", ignoreCase = true)
    val isHls: Boolean get() = type.equals("HLS", ignoreCase = true)
    val isFlow: Boolean get() = headers != null
}

File: Dash MPD Player/mitube.service/Models/Channel.cs
using System.Text.Json.Serialization;

namespace mitube.service.Models;

public class Channel
{
    [JsonPropertyName("name")]
    public string Name { get; set; } = "";

    [JsonPropertyName("url")]
    public string Url { get; set; } = "";

    [JsonPropertyName("type")]
    public string Type { get; set; } = "";

    [JsonPropertyName("drm_license_uri")]
    public string DrmLicenseUri { get; set; } = "";

    [JsonPropertyName("icono")]
    public string Icono { get; set; } = "";

    [JsonPropertyName("headers")]
    public Dictionary<string, string>? Headers { get; set; }

    public (string keyId, string key)? ParseClearkeyLicense()
    {
        if (string.IsNullOrEmpty(DrmLicenseUri)) return null;
        var uri = new Uri(DrmLicenseUri);
        var query = uri.Query.TrimStart('?').Split('&')
            .Select(p => p.Split('=', 2))
            .Where(p => p.Length == 2)
            .ToDictionary(p => Uri.UnescapeDataString(p[0]), p => Uri.UnescapeDataString(p[1]), StringComparer.OrdinalIgnoreCase);
        query.TryGetValue("keyid", out var keyId);
        query.TryGetValue("key", out var key);
        if (!string.IsNullOrEmpty(keyId) && !string.IsNullOrEmpty(key))
            return (keyId, key);
        return null;
    }
}
